app/helpers/advanced_helper.rb

```diff
@@ -26,18 +26,17 @@ def advanced_key_value
 
   # carries over original search field and original guided search fields if user switches to guided search from regular search
   def guided_field(field_num, default_val)
-    if field_num == :f1 && params[:f1].nil? && params[:f2].nil? && params[:f3].nil? && params[:search_field] && search_fields_for_advanced_search[params[:search_field]]
-      return search_fields_for_advanced_search[params[:search_field]].key || default_val
-    end
+    return search_fields_for_advanced_search[params[:search_field]].key || default_val if field_num == :f1 && params[:f1].nil? && params[:f2].nil? && params[:f3].nil? && params[:search_field] && search_fields_for_advanced_search[params[:search_field]]
```

Comment:
Could we move the clause of this conditional to a method that's called what it's checking for?

Reply:
Done.